src/Kconfig

```diff
@@ -142,7 +142,7 @@ config WANT_SENSORS
     depends on HAVE_GPIO_I2C || HAVE_GPIO_SPI
 config WANT_LIS2DW
     bool "Support lis2dw 3-axis accelerometer"
-    depends on HAVE_GPIO_SPI
+    depends on HAVE_GPIO_SPI || HAVE_GPIO_I2C
```

Comment:
In practice, every real-world board with spi also has i2c (and vice-versa).  So, adding lots of ifdefs just makes the code harder to maintain.

If you're really concerned (and I don't think you should be) then you could change all the code to use run-time if statements (eg, `if (CONFIG_HAVE_GPIO_I2C && ax->bus_type == I2C_BUS) ...`

-Kevin

Reply:
I am fairly certain runtime checks will not work here, since there does not exist both an I2C and SPI implementation for each chip. I expect the build would fail, complaining either about `spi_transfer` or `i2c_read` in these cases. I think the preprocessor conditionals are not that complex in this case, because at a high level they just selectively choose to add a set of conditional bus type checks guarding an implementation for either SPI or I2C, depending on what the chip has available.

Reply:
Run-time checks should work as gcc is very good about eliminating dead code - it'll interpret the conditionals as `if (0 && ...)`, remove the entire branch, and the linker will never see the calls.  This mechanism is used through the Klipper code.

-Kevin

Reply:
Actually, you'd also have to define a dummy `struct i2c_config` in src/ar100/gpio.h and src/generic/gpio.h .  See the dummy 'struct spi_config' in src/generic/gpio.h for an example.

-Kevin

Reply:
Have a look at the current updates, migrated to runtime checks.